packages/template-retail-react-app/jest-setup.js

```diff
@@ -9,6 +9,63 @@ const mockConfig = require(path.join(__dirname, 'config/mocks/default.js'))
 require('raf/polyfill') // fix requestAnimationFrame issue with polyfill
 require('@testing-library/jest-dom/extend-expect')
 const {Crypto} = require('@peculiar/webcrypto')
+const {setupServer} = require('msw/node')
+const {rest} = require('msw')
+const {
+    mockCategory,
+    mockedRegisteredCustomer,
+    exampleTokenReponse
+} = require('./app/commerce-api/mock-data')
+
+/**
+ * Set up an API mocking server for testing purposes.
+ * This mock server includes the basic oauth flow endpoints.
+ */
+export const setupMockServer = () => {
+    return setupServer(
+        rest.post('*/oauth2/authorize', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.get('*/oauth2/authorize', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.post('*/oauth2/login', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        ),
+        rest.get('*/oauth2/logout', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(exampleTokenReponse))
+        ),
+        rest.get('*/customers/:customerId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        ),
+        rest.post('*/sessions', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.post('*/oauth2/token', (req, res, ctx) =>
+            res(
+                ctx.delay(0),
+                ctx.json({
+                    customer_id: 'test',
+                    access_token: 'testtoken',
+                    refresh_token: 'testrefeshtoken',
+                    usid: 'testusid',
+                    enc_user_id: 'testEncUserId',
+                    id_token: 'testIdToken'
+                })
+            )
+        ),
+        rest.get('*/categories/:categoryId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockCategory))
+        ),
+        rest.post('*/baskets/actions/merge', (req, res, ctx) => res(ctx.delay(0), ctx.status(200)))
+    )
+}
+
+global.server = setupMockServer()
```

Comment:
There is a guaranteed known state. It is `setupMockServer` (and to @kevinxh 's point, we should have a global catch to prevent any `get|put|patch` etc. requests from touching the network).

I'm going to disagree here, our `setupMockServer` implementation is unnecessary boilerplate and with the introduction of the `<CategoriesProvider />` to make an API call, we would pretty much have to add `setupMockServer` to every test file. Thus we are doing it globally. 

We need to merge this and not rework it forever, so I'm going to vote against localizing / non-globalizing this

Reply:
As long as we can guarantee a known state per test, then that's fine. Should still switch to `import`/`export`, though, unless there's weird jest nonsense going on.

Reply:
I attempted to switch to `import` / `export`, but things are not working for some reasons I couldn't figure out. Let's make this a iterative change and we will need to take a closer look at those tests.